src/main/java/org/wso2/carbon/inbound/kafka/KafkaMessageConsumer.java

```diff
@@ -0,0 +1,444 @@
+            }
+
+            if (properties.getProperty(KafkaConstants.SASL_KERBEROS_TICKET_RENEW_JITTER) != null) {
+                properties.setProperty(KafkaConstants.SASL_KERBEROS_TICKET_RENEW_JITTER,
+                        properties.getProperty(KafkaConstants.SASL_KERBEROS_TICKET_RENEW_JITTER));
+            }
+
+            if (properties.getProperty(KafkaConstants.SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR) != null) {
+                kafkaProperties.put(KafkaConstants.SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR,
+                        properties.getProperty(KafkaConstants.SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_CIPHER_SUITES) != null) {
+                properties.setProperty(KafkaConstants.SSL_CIPHER_SUITES,
+                        properties.getProperty(KafkaConstants.SSL_CIPHER_SUITES));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM,
+                        properties.getProperty(KafkaConstants.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_CIPHER_SUITES) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_CIPHER_SUITES,
+                        properties.getProperty(KafkaConstants.SSL_CIPHER_SUITES));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_KEYMANAGER_ALGORITHM) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_KEYMANAGER_ALGORITHM,
+                        properties.getProperty(KafkaConstants.SSL_KEYMANAGER_ALGORITHM));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_SECURE_RANDOM_IMPLEMENTATION) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_SECURE_RANDOM_IMPLEMENTATION,
+                        properties.getProperty(KafkaConstants.SSL_SECURE_RANDOM_IMPLEMENTATION));
+            }
+
+            if (properties.getProperty(KafkaConstants.SSL_TRUSTMANAGER_ALGORITHM) != null) {
+                kafkaProperties.put(KafkaConstants.SSL_TRUSTMANAGER_ALGORITHM,
+                        properties.getProperty(KafkaConstants.SSL_TRUSTMANAGER_ALGORITHM));
+            }
+
+            consumer = new KafkaConsumer<>(kafkaProperties);
+        } catch (Exception e) {
+            throw new SynapseException("Failed to construct kafka consumer", e);
+        }
+
+        String[] topicsArray = topic.split(",");
+        consumer.subscribe(Arrays.asList(topicsArray));
+
+        try {
+            while (true) {
+                ConsumerRecords<byte[], byte[]> records = consumer.poll(Long.parseLong(pollTimeout));
+                for (ConsumerRecord record : records) {
+                    msgCtx = createMessageContext();
```

Comment:
We need to set the properties in the messageContext to get the value from sequence for that we need to pass the messageContext to https://github.com/hariss63/esb-inbound-kafka/blob/6f5e6ccd462dc964a0e175a4be1707c48306c7f8/src/main/java/org/wso2/carbon/inbound/kafka/KafkaMessageConsumer.java#L368. Because we don't have a way to pass the message context to injectMessage method in GenericPollingConsumer.